### PR TITLE
OCPBUGS-38006: Use CAPO v0.10 and API v1beta1

### PR DESCRIPTION
**Pull request by mdbooth**
This fixes a crashloop when deploying on OSP because the version of CAPO we deploy no longer serves v1alpha7.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/cluster-capi-operator
 
-go 1.21
+go 1.22
```
Comment (mdbooth) [inline go.mod:3]: We have a Go builder for 4.17, so I don't think it's worth jumping through hoops to try to avoid this bump. CAPO v0.10 uses go 1.22.

**Review by mandre — COMMENTED** [review]
Last CI run shows that [cluster-capi-operator is now stable](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_cluster-capi-operator/203/pull-ci-openshift-cluster-capi-operator-main-e2e-openstack-capi-techpreview/1832072079818625024/artifacts/e2e-openstack-capi-techpreview/gather-extra/artifacts/pods/openshift-cluster-api_cluster-capi-operator-59655d9f88-v7v9f_cluster-capi-operator.log).

/lgtm

**Comment by mandre** [comment]
The e2e-aws-ovn-techpreview job failed but appears to be stable otherwise. Let's try again.
/retest

**Pull request merged**